src/app/components/cart/cart.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
+import { Card } from '../products/models/card.interface';
+import { PaginationService } from '../products/core/pagination/pagination.service';
+import { CartService } from '../products/core/cart/cart.service';
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.scss']
+})
+export class CartComponent implements OnInit {
+
+  
+  cart:Card[] = []
+  numberOfPages = 1
+  currentPageProducts:Card[] = []
+  constructor(private CartService: CartService, private PaginationService: PaginationService) { }
+
+  ngOnInit(): void {
+    this.cart = this.CartService.getCart()
+    this.PaginationService.setProductForCurrentPage(this.cart)
+    this.currentPageProducts = this.PaginationService.getProductForCurrentPage() 
+    this.PaginationService.setNumberOfPages(this.cart.length )
+    this.numberOfPages = this.PaginationService.getNumberOfPages()
+
+  }
+  ngDoCheck():void{
```

Comment:
Looks like this function not in use. Also you have duplicated code fragment, same logic in ngOnInit function